### ISSUE TITLE
Promise returned by share dialog won't resolve if cancelled on iOS

### ISSUE THREAD

**Issue by v-fernandez** [facebook/react-native#26809]
On iOS, the share text dialog won't resolve the returned promise when the dialog is dismissed by pressing the Cancel button or by pressing outside the dialog. Other share options resolve the promise as expected.

```typescript
shareLink = async () => {
  await Share.share({ message: 'message-to-share' }) // Hangs if the dialog is dismissed by the Cancel button or by pressing outside the dialog
}
```

React Native version: 0.61.2

## Steps To Reproduce

1. Open share dialog with `Share.share(content, options)`
2. Dismiss the dialog by pressing Cancel or pressing outside the dialog.

Describe what you expected to happen: The promise returned by `Share.share(content, options)` should be resolved when the dialog is dismissed.

Snack, code example, screenshot, or link to a repository:

**Comment by v-fernandez** [comment]
To add more info to this. I believe the bug might be here: https://github.com/facebook/react-native/blob/master/React/CoreModules/RCTActionSheetManager.mm#L183

Changing the `else if` to an `else` has resolved my issue.

**Comment by cdunkel** [comment]
@v-fernandez, do you know when this fix will be available, if it's not already?

**Comment by v-fernandez** [comment]
@cdunkel I'm not sure when this will be available, but in the meantime, you can use [patch-package](https://github.com/ds300/patch-package#readme) to make the fix available for you in the version you are using.

**Comment by metawort** [comment]
Why is the fix still not merged? I still encounter this issue in 0.61.5

**Comment by v-fernandez** [comment]
@metawort this will be available on v0.62.0. See https://github.com/facebook/react-native/pull/26842#issuecomment-570528308
